src/map.cpp

```diff
@@ -6038,6 +6038,44 @@ bool map::sees( const tripoint &F, const tripoint &T, const int range, int &bres
     return visible;
 }
 
+int map::obstacle_coverage( const tripoint &loc1, const tripoint &loc2 )
+{
+    if( move_cost( loc2 ) > 2 && !has_flag_ter( TFLAG_FLAT, loc2 ) ) { // Can't hide if you are standing on something.
```

Comment:
`move_cost` is a rather fragile assumption for "standing on something".

Reply:
I think it's okay, move_cost increases when there's furniture or terrain obstacle on a tile, the only cases where this approach might fail is sand, long grass or rails tiles, that is why I added the check for `FLAT` flag.